File: week10/in-class-exercises/python/q2a.py
# Nicholas Ong - Q2a: Books
def get_books_cheaper_than(input_file_name, price_limit, output_file_name):
    infile = open(f"../src/{input_file_name}", "r")
    outfile = open(output_file_name, "w")

    result = []
    lines = infile.readlines()
    for line in lines:
        a = line.strip().split()
        try:
            if float(a[-1][1:]) < price_limit:
                result.append(line)
        except Exception as e:
            print(e)
    outfile.writelines(result)

    infile.close()
    outfile.close()

def main():
    test_cases = [["books-1.txt","books-1-output.txt", 15.0], ["books-2.txt", "books-2-output.txt", 7.0]]
    for input_file_name, output_file_name, price_limit in test_cases:
        get_books_cheaper_than(input_file_name, price_limit, output_file_name)

if __name__ == "__main__":
    main()

File: week10/in-class-exercises/cpp/q2a.cpp
#include <fstream>
#include <iostream>
#include <string>
#include <vector>

void get_books_cheaper_than ()
{
  std::ifstream infile{"../src/books-1.txt",std::ios::out};
  while (infile)
  {
    std::string strInput;
    std::vector<std::string>line(3);
    for(std::string temp; std::getline(infile,strInput,'\t');)
    {
      line.push_back(temp);
      std::cout<<temp<<'\n';
    }
  }
  return;
}

int main()
{
  get_books_cheaper_than();
}
